Add extraction process with ogr2ogr and fix creation of ZIP archive

Vampouille opened the pull request:
This PR also :
* introduce database in docker composition
* use gdal 2.1.0 and python 3.5 from debian stretch

Vampouille committed:
Add database in composition

Vampouille committed:
Add gdal-bin 2.1.0 from debian stretch repo and reorganize Dockerfile

Vampouille committed:
add extraction with ogr2ogr on CLI with subprocess.Popen()

Vampouille committed:
Fix creation of ZIP archive

fvanderbiest commented:
you mean ogr2ogr ? ;-P

fvanderbiest commented on celery/extractions.py:
```diff
@@ -14,13 +20,109 @@
+    :return: None
+    """
+    p = Popen(args, stdout=PIPE, stderr=PIPE)
+    p.wait()
```
I'm wondering if we should specify a timeout argument to wait().
https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Vampouille replied:
yes good idea

Vampouille replied:
10 min ?

fvanderbiest replied:
No idea. 
I guess a new ENV var for this would allow us to be more flexible.

fvanderbiest commented on celery/extractions.py:
```diff
@@ -14,13 +20,109 @@
+    return res
+
+
+def export_schema_to_shapefile_or_mapfile(year, proj, output_dir, format, conn, pg_connect_string):
```
Please rename to `export_schema_to_shapefile_or_mapinfo` (no mapfile involved here ;-)

fvanderbiest commented on celery/extractions.py:
```diff
@@ -14,13 +20,109 @@
+    :param proj: output projection
+    :param output_dir: folder where files will be writen
+    :param conn: a psycopg connection instance
+    :param pg_connect_string: a string that contains option to connect to database with ogr2ogr. 'schema' option will
```
schemas, not schema

fvanderbiest commented on celery/extractions.py:
```diff
@@ -14,13 +20,109 @@
+    be added
+    :return: None
+    """
+    with open(join(output_dir, "foncier_%s.sql"  % year), 'wb') as f:
```
`wb` ? no binary involved here, just plain text, no ?

Vampouille replied:
There are some UTF-8 characters that are easier to process with binary flag. I have to add this flag after testing against test data.

fvanderbiest commented on celery/extractions.py:
```diff
@@ -14,13 +20,109 @@
 def do(year, format, proj, email, cities):
-    tmpdir = tempfile.mkdtemp(dir = FONCIER_EXTRACTS_DIR, prefix = 'foncier_{0}_{1}_{2}_{3}-'.format(year, format, proj, do.request.id))
+
+    extraction_id = 'foncier_{0}_{1}_{2}_{3}'.format(year, format, proj, do.request.id)
```
`foncier_{0}_{1}_{2}_{3}-` is nicer IMO, but no blocker.

Vampouille replied:
Ok, I will use this format.

fvanderbiest commented:
Finished my review, LGTM.
Thanks for the great code.

fvanderbiest commented on docker-compose.yml:
```diff
@@ -20,11 +20,13 @@ worker:
   environment:
     - CELERY_BROKER_URL=redis://redis:6379/0
     - CELERY_RESULT_BACKEND=redis://redis:6379/0
+    - PG_CONNECT_STRING=host=database port=5432 user=geoserver dbname=geoserver password=geoserver
```
`sslmode=require` ? I'm not sure about it.

Vampouille replied:
not needed for docker setup

fvanderbiest commented:
Oops, I completely forgot that we should filter the results on the cities list !
Maybe you planned it for a subsequent PR ?

Vampouille commented:
> Maybe you planned it for a subsequent PR ?

Yes

Vampouille committed:
Small fixes

Rename export_schema_to_shapefile_or_mapfile() function with 'mapinfo'
Fix typos
Fix usage of PG_CONNECT_STRING/pg_connect_string
